File: python/剑指offer/面试题18. 删除链表的节点.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        cur,= head
        if cur.val == val:
            return head.next
        pre = cur
        cur = cur.next

        while cur:
            if cur.val == val:
                pre.next = cur.next
                cur = cur.next
            else:
                pre = pre.next
                cur = cur.next
        return head

File: go/剑指offer/面试题18. 删除链表的节点.go
package main

func deleteNode(head *ListNode, val int) *ListNode {
	cur := head
	if cur.Val == val {
		return head.Next
	}
	pre := head
	cur = cur.Next
	for cur != nil {
		if cur.Val == val {
			pre.Next = cur.Next
			cur = cur.Next
		} else {
			pre = pre.Next
			cur = cur.Next
		}
	}
	return head
}
